Makefile

```diff
@@ -175,47 +144,29 @@ shell: build-dirs build-env
 		$(BUILDER_IMAGE) \
 		/bin/sh $(CMD)
 
-DOTFILE_IMAGE = $(subst :,_,$(subst /,_,$(IMAGE))-$(VERSION))
-
-all-containers:
-	$(MAKE) container
-	$(MAKE) container BIN=velero-restic-restore-helper
-
-container: local-arch .container-$(DOTFILE_IMAGE) container-name
-.container-$(DOTFILE_IMAGE): _output/bin/$(GOOS)/$(GOARCH)/$(BIN) $(DOCKERFILE)
-	@cp $(DOCKERFILE) _output/.dockerfile-$(BIN)-$(GOOS)-$(GOARCH)
-	@docker build --pull -t $(IMAGE):$(VERSION) -f _output/.dockerfile-$(BIN)-$(GOOS)-$(GOARCH) _output
-	@docker images -q $(IMAGE):$(VERSION) > $@
-
-container-name:
+container-build-env:
+	@docker buildx build \
+	--target=builder-env \
+	--build-arg=PKG=$(PKG) \
+	--build-arg=VERSION=$(VERSION) \
+	--build-arg=GIT_SHA=$(GIT_SHA) \
+	--build-arg=GIT_TREE_STATE=$(GIT_TREE_STATE) \
+	-f Dockerfile .
+
+container:
+	@docker buildx build --pull \
+	--output=type=registry \
```

Comment:
Is there a way to push images outside of the auto-push?
This is to avoid accidental pushes to the `velero` registry and also most of our dev work happens locally without an image registry.

Reply:
There is not right now, it will be possible once https://github.com/moby/moby/pull/38738 has been merged and released.

As a work around in the mean time, I have added a check to the container build recipe to prevent accidental unintended pushes and added a local build option for the container images, which will only build the velero and velero-restic-restore-helper images for the arch you are running it from. This will allow a developer to locally build a docker image so it can be tested.

Let me know if this solution works for you.

Reply:
That is reasonable to me!